src/mam/mod_mam_rdbms_arch.erl

```diff
@@ -102,6 +103,7 @@ start_hooks(Host, _Opts) ->
     ejabberd_hooks:add(mam_archive_size, Host, ?MODULE, archive_size, 50),
     ejabberd_hooks:add(mam_lookup_messages, Host, ?MODULE, lookup_messages, 50),
     ejabberd_hooks:add(mam_remove_archive, Host, ?MODULE, remove_archive, 50),
+    ejabberd_hooks:add(remove_domain, Host, ?MODULE, remove_domain, 50),
```

Comment:
please, make a `hooks/0` function that returns a list of hooks and use `ejabbed_hooks:add/1` and `ejabberd_hooks:delete/1` functions to start/stop module